mqtt/mqtt-broker/src/configuration.rs

```diff
@@ -77,16 +80,20 @@ impl SessionConfig {
         }
     }
 
-    pub fn max_inflight_messages(&self) -> usize {
-        self.max_inflight_messages
+    pub fn max_inflight_messages(&self) -> Option<NonZeroUsize> {
+        NonZeroUsize::new(self.max_inflight_messages)
```

Comment:
It looks like it does not handle in the way we need. We can't just define field as `Option<NonZeroUsize>` since it uses Option<T> to denote presence of the field, not that it has invalid value in the json. As an alternative - create newtype and implement custom deserializer. But IMHO it is unnecessary here.

Reply:
Agreed. Thanks for explanation!